apps/opencs/model/world/commands.cpp

```diff
@@ -199,8 +199,10 @@ CSMWorld::ModifyCommand::ModifyCommand (QAbstractItemModel& model, const QModelI
     if (mIndex.parent().isValid())
     {
         CSMWorld::IdTree* tree = dynamic_cast<CSMWorld::IdTree*>(mModel);
-
-        assert(tree != nullptr);
+        if (tree == nullptr)
```

Comment:
This is very verbose (and all the code repetition makes it worse. I suggest to cast via reference. This will allow the build in C++ mechanics to take care of everything, e.g.:

CSMWorld::IdTree* tree = &dynamic_cast<CSMWorld::IdTree&>(*mModel);

If this cast fails it will automatically throw a suitable exception.

Reply:
Ok.